src/main/kotlin/io/github/mojira/arisa/infrastructure/jira/Mappers.kt

```diff
@@ -172,14 +173,34 @@ fun JiraComment.toDomain(
 
 fun JiraUser.toDomain(jiraClient: JiraClient) = User(
     name, displayName,
-    ::getUserGroups.partially1(jiraClient).partially1(name)
+    ::getUserGroups.partially1(jiraClient).partially1(name),
+    ::isNewUser.partially1(jiraClient).partially1(name)
 )
 
 private fun getUserGroups(jiraClient: JiraClient, username: String) = getGroups(
     jiraClient,
     username
 ).fold({ null }, { it })
 
+private fun isNewUser(jiraClient: JiraClient, username: String): Boolean {
+    val escapedUserName = if (username.contains('\'')) """\\"$username\\"""" else "'$username'"
+    val commentJql = """issueFunction IN commented("by $escapedUserName before -24h")"""
+
+    val oldCommentsExist = try {
+        jiraClient.countIssues(commentJql) > 0
+    } catch (_: JiraException) { false }
+
+    if (oldCommentsExist) return false
+
+    val reportJql = """project != TRASH AND reporter = '${username.replace("'", "\\'")}' AND created < -24h"""
```

Comment:
No, because the issue function may fail, [e.g. if the name exceeds a certain length](https://bugs.mojang.com/browse/MCTEST-113?focusedCommentId=983363&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-983363). If this was a single OR query this check would be trivial to circumvent.

This is also why I don't directly assume that the user is not a new user if this query fails.